# HG changeset patch
# Parent  cdd7a249466bd87fa2fd3030d3238d58657f004a
Add a docs CMake target to easily call sphinx-build

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,9 +8,9 @@
 SET(CPACK_PACKAGE_VERSION_PATCH "0")
 SET(LIGHTSD_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
 
+MESSAGE(STATUS "lightsd version: ${LIGHTSD_VERSION}")
 MESSAGE(STATUS "CMake version: ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}")
 MESSAGE(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
-MESSAGE(STATUS "lightsd version: ${LIGHTSD_VERSION}")
 MESSAGE(STATUS "Target system: ${CMAKE_SYSTEM_NAME} ${CMAKE_SYSTEM_VERSION} ${CMAKE_SYSTEM_PROCESSOR}")
 MESSAGE(STATUS "Install prefix: ${CMAKE_INSTALL_PREFIX}")
 MESSAGE(STATUS "Source directory: ${LIGHTSD_SOURCE_DIR}")
@@ -24,6 +24,7 @@
 # TODO: we need at least 2.0.19-stable because of the logging defines
 FIND_PACKAGE(Event2 REQUIRED COMPONENTS core)
 FIND_PACKAGE(Endian REQUIRED)
+FIND_PACKAGE(Sphinx)
 
 INCLUDE(CheckFunctionExists)
 INCLUDE(CheckVariableExists)
@@ -88,6 +89,7 @@
 ADD_SUBDIRECTORY(compat)
 ADD_SUBDIRECTORY(core)
 ADD_SUBDIRECTORY(lifx)
+
 # 2.8.11 is the first version with TARGET_INCLUDE_DIRECTORIES:
 IF (CMAKE_VERSION VERSION_GREATER 2.8.10)
     CONFIGURE_FILE(
@@ -102,6 +104,10 @@
     )
 ENDIF ()
 
+IF (SPHINX_FOUND)
+    ADD_SUBDIRECTORY(docs)
+ENDIF ()
+
 INSTALL(
     FILES COPYING README.rst docs/protocol.rst
     DESTINATION share/doc/lightsd
diff --git a/CMakeScripts/FindSphinx.cmake b/CMakeScripts/FindSphinx.cmake
new file mode 100644
--- /dev/null
+++ b/CMakeScripts/FindSphinx.cmake
@@ -0,0 +1,7 @@
+FIND_PROGRAM(
+    SPHINX_EXECUTABLE
+    NAMES sphinx-build sphinx-build2
+    DOC "Path to sphinx-build executable"
+)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(Sphinx DEFAULT_MSG SPHINX_EXECUTABLE)
diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
new file mode 100644
--- /dev/null
+++ b/docs/CMakeLists.txt
@@ -0,0 +1,9 @@
+CONFIGURE_FILE(conf.py.in "${CMAKE_CURRENT_BINARY_DIR}/conf.py")
+
+ADD_CUSTOM_TARGET(
+    docs
+    COMMAND "${SPHINX_EXECUTABLE}" -v -W -b html -c "${CMAKE_CURRENT_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}" _build
+    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+    COMMENT "Building the documentation in ${CMAKE_CURRENT_BINARY_DIR}/_build with sphinx-build"
+    VERBATIM
+)
diff --git a/docs/_static/.dummy b/docs/_static/.dummy
new file mode 100644
diff --git a/docs/conf.py b/docs/conf.py.in
rename from docs/conf.py
rename to docs/conf.py.in
--- a/docs/conf.py
+++ b/docs/conf.py.in
@@ -24,12 +24,14 @@
 # -- General configuration ------------------------------------------------
 
 # If your documentation needs a minimal Sphinx version, state it here.
-#needs_sphinx = '1.0'
+needs_sphinx = '1.2.0'
 
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = []
+extensions = ['sphinx.ext.extlinks']
+
+extlinks = {'gh': ('https://github.com/lopter/lightsd/issues/%s', 'GH-')}
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
@@ -52,9 +54,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '1.1.0'
+version = '@LIGHTSD_VERSION@'
 # The full version, including alpha/beta/rc tags.
-release = '1.1.0'
+release = '@LIGHTSD_VERSION@'
 
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
@@ -128,7 +130,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+html_static_path = [os.path.join('@CMAKE_CURRENT_SOURCE_DIR@', '_static')]
 
 # Add any extra paths that contain custom files (such as robots.txt or
 # .htaccess) here, relative to this directory. These files are copied
